fabric-command-api-v1/src/main/java/net/fabricmc/fabric/impl/command/client/ClientCommandInternals.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.command.client;
+
+import static net.fabricmc.fabric.api.client.command.v1.ClientCommandManager.DISPATCHER;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.mojang.brigadier.AmbiguityConsumer;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.builder.ArgumentBuilder;
+import com.mojang.brigadier.exceptions.BuiltInExceptionProvider;
+import com.mojang.brigadier.exceptions.CommandExceptionType;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.tree.CommandNode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.command.CommandException;
+import net.minecraft.text.Text;
+import net.minecraft.text.Texts;
+import net.minecraft.text.TranslatableText;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource;
+
+@Environment(EnvType.CLIENT)
+public final class ClientCommandInternals {
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final char PREFIX = '/';
```

Comment:
I think there was already a discussion on this in the discord if these commands should have a different/customizable prefix. If not, it would be nice to have customizable prefixes and if the API is going through the trouble of adding client commands, we might as well have commands be able to have custom prefixes. (obviously default to `/`).

Reply:
Ah, it seems that modmuss50 already gave some input on this, but I think that having two registration methods, one for a default prefix and one for a custom one is better. I feel like the main reason to have a different prefix is to think about how it has a different function than `/`,

Reply:
Well, I had custom prefixes working quite nicely originally. I removed them because modmuss didn't want them in Fabric API (to keep the client commands closer to vanilla's server commands).

Reply:
Which seems a bit weird, because it doesn't seem to handle conflict very well.

Like if a mod register a command `a` and the server also has a command `a` what happens?

Custom prefixes could partially fix this issue, but at the same time some custom prefixes might be weird/annoying.

Client commands having the same prefix as everything else doesn't seem very user-friendly in the end.

Reply:
> Like if a mod register a command `a` and the server also has a command `a` what happens?

I actually haven't tested this, but with the current impl the client command should override the server command. It might be more sensible to reverse that behaviour.

@modmuss50 Should we reconsider custom prefixes for compatibility with servers? I'm not sure whether there should use only `/`, only some client-specific character (say, `#`) or to have the full prefix support again.

Reply:
I don't know if it's possible, but for the conflict situation I would like to see the Bukkit behavior which is to add `namespace:` at the start of the command.

Reply:
`namespace:command` is fairly pointless when you can do `namespace command`

Reply:
A custom prefix is out of scope, IMO the best solution is to eventually:
- give precedence to server side commands
- use `/fcc x` to always run the client command `/x`

This will resolve the ambiguity cleanly, as long as nothing else claims `/fcc`, which me might exclude from the server precedence.

Reply:
I feel like /fcc should just be a shortcut to a full /fabricclientcommand or whatever, much like `/tp` is a short alias for `/teleport`. It lets people find it if they don't know the acronym, and if they do then they can type it quickly.